snuba/query/parser/__init__.py

```diff
@@ -92,18 +111,19 @@ def _parse_query_impl(body: MutableMapping[str, Any], dataset: Dataset) -> Query
         alias = alias if alias else None
 
         aggregate_exprs.append(
-            parse_aggregation(aggregation_function, column_expr, alias)
+            SelectedExpression(
+                name=alias,
+                expression=parse_aggregation(aggregation_function, column_expr, alias),
+            )
         )
 
-    groupby_exprs = [
-        parse_expression(tuplify(group_by))
-        for group_by in to_list(body.get("groupby", []))
-    ]
-    select_exprs = [
-        parse_expression(tuplify(select)) for select in body.get("selected_columns", [])
-    ]
+    groupby_exprs = build_selected_expressions(to_list(body.get("groupby", [])))
 
-    selected_cols = groupby_exprs + aggregate_exprs + select_exprs
+    select_clause = (
+        groupby_exprs
+        + aggregate_exprs
```

Comment:
I think these names are a bit misleading now since these are not actually expressions (especially since the group by items can't be passed directly to the `Query` constructor without extracting the expression attribute.)